### DISCUSSION TITLE
Weekly Forecasting/Prediction of Energy Data

### DISCUSSION THREAD

**Discussion by AntonCiu** [online-ml/river#847] [Q&A] [answered]
Hi,

I am currently getting my head around this library and trying to figure out if it will suit what I am doing.

Basically I am trying to forecast a week of Electricity Consumption data in 30 minute intervals using weather data. I have had some success in building a model to predict_one (The next timestep) of energy data fairly accurately. 

I want to now evolve this model to always predict a week of data and constantly update with new data received.

How does the predict_many & learn_many methods work I haven't been able to find much in the API reference. Will these functions allow me to predict a week of data?

Is it possible to predict this far ahead? Is 1 day of data more realistic? 

Thanks,
Anton

**Comment by MaxHalford** [comment, marked as answer]
> Basically I am trying to forecast a week of Electricity Consumption data in 30 minute intervals using weather data

That can be tricky, because you need to use weather data estimates, which are themselves predictions. Hope you're aware of it.

> I want to now evolve this model to always predict a week of data and constantly update with new data received.

I would suggest starting with the [HoltWinters](https://riverml.xyz/latest/api/time-series/HoltWinters/) method. It should give decent results if your data exhibits some periodicity.

> Is it possible to predict this far ahead? Is 1 day of data more realistic?

That depends on your data, I can't say without looking at it :)

**Reply by AntonCiu** [reply]
Thank you for the reply!

Yes that is one limiting factor of this task that I am aware of, I will look into HoltWinters method and see if it is what I am looking for. 

Much appreciated,
Anton

**Reply by dberardo-com** [reply]
@AntonCiu if you are still seeking for help, feel free of sharing a sample dataset for this problem, so we can try figuring out a solution together. I did this in the past for the WaterFlow dataset and that resulted in a new feature release.

Is the weather forecast a given input (measured) in your problem or is it also to be forecasted?

> How does the predict_many & learn_many methods work I haven't been able to find much in the API reference. Will these functions allow me to predict a week of data?

have you tried to generate a vector of length 30minutes*1week and pick the last value of the array? would that work in your case?

**Comment by dberardo-com** [comment]
@MaxHalford somewhat related question: has anyone ever started a discussion on the matter "prediction/confidence intervals"? i was going through some suggested sources ( https://otexts.com/fpp2/prediction-intervals.html ) and it seems like some assumption could be made for them, also incrementally.

Should i start a new discussion?

**Reply by MaxHalford** [reply]
No we haven't looked into it. It would be nice to add it. Just need to figure out the correct API.
